hudi-common/src/main/java/org/apache/hudi/common/util/ReflectionUtils.java

```diff
@@ -85,11 +87,14 @@ public static <T extends HoodieRecordPayload> T loadPayload(String recordPayload
    * Creates an instance of the given class. Use this version when dealing with interface types as constructor args.
    */
   public static Object loadClass(String clazz, Class<?>[] constructorArgTypes, Object... constructorArgs) {
-    try {
-      return getClass(clazz).getConstructor(constructorArgTypes).newInstance(constructorArgs);
-    } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
-      throw new HoodieException("Unable to instantiate class " + clazz, e);
-    }
+    return newInstanceUnchecked(getClass(clazz), constructorArgTypes, constructorArgs);
+  }
+
+  /**
+   * Creates an instance of the given class. Constructor arg types are inferred.
+   */
+  public static Object loadClass(String clazz, Object... constructorArgs) {
+    return newInstanceUnchecked(getClass(clazz), constructorArgs);
```

Comment:
this is affecting a lot of code paths. can't we make a specialized one just for this patch and leave the original as is? we should mitigate the risk here.

Reply:
I don't even see this `newInstanceUnchecked` was used in this patch. can we revert the change? or create a special one if you need it here?

Reply:
It was used initially, but had to steer away b/c of Scala limitations (Scala enums are problematic from the reflection standpoint). 

This code is executed in every writing test, and it just moves implementation from one method to another to expose a new method, i don't think we need to worry about this one in particular.

Reply:
On a second thought, it's a fair point that there's no reason to keep it in this PR, since it's not used in it. Removed from it.